cla_backend/apps/reports/tests/test_views.py

```diff
@@ -0,0 +1,33 @@
+from mock import patch, MagicMock, mock_open
+from django.test import TestCase
+from reports.views import download_file
+from django.conf import settings
+
+
+class DownloadFileTestCase(TestCase):
+    @patch("reports.views.get_s3_connection")
+    def test_download_no_aws(self, mock_s3):
+        # mock pythons open()
+        with patch("__builtin__.open", mock_open(read_data="data")) as mock_file:
+            mock_request = MagicMock()
+            # if file_name contains string "schedule"
+            # delete from the database doesn't occur.
+            file_name = "scheduled"
+            settings.AWS_REPORTS_STORAGE_BUCKET_NAME = ""
+            settings.DEBUG = True
+            # download_file(request, file_name="", *args, **kwargs):
+            download_file(mock_request, file_name)
+            assert open("path/to/open").read() == "data"
```

Comment:
I am not sure which part of reports.views.download_file this is testing?

Reply:
Ah yes, this can actually be removed as it proves nothing and is not needed. It doesn't prove anything now I have a much better understanding behind writing unit tests in Python.

So all this is doing is asserting when you call the Open method it reads the mock read_data.... Then the assert after just checks it was called with that path. It has nothing to do with the test... So removed.

Good catch, it's redundant code.